splice/web/api/content_upload.py

```diff
@@ -0,0 +1,209 @@
+import hashlib
+import os
+import base64
+import json
+import requests
+import zipfile
+import tempfile
+
+from furl import furl
+from mohawk import Sender
+from boto.s3.key import Key
+from splice.web.api.s3_common import setup_s3, MIME_EXTENSIONS
+from splice.environment import Environment
+
+
+env = Environment.instance()
+_ORIGINAL_NAME = "original.zip"
+
+
+def _get_original_content(name, version, bucket):
+    original = tempfile.NamedTemporaryFile(prefix='splice', suffix='.zip')
+    s3_key = "{0}/v{1}/{2}".format(name, version, _ORIGINAL_NAME)
+    k = Key(bucket)
+    k.key = s3_key
+    try:
+        k.get_file(original)
+    except:
+        original.close()
+        return None
+    else:
+        return original
+
+
+def resign_content(name, version):
+    """Re-sign a content till the given version. It downloads the original
+    content file from S3, and only re-signs the assets in the manifest. Also,
+    it won't change the version of the content.
+
+    Params:
+        name: name of the target content
+        version: the current version of the target content
+    Returns: None
+    Exception: an exception will be raised if any error happens
+    """
+    bucket, headers = setup_s3(bucket="content")
+    for v in range(1, version + 1):
+        original = _get_original_content(name, v, bucket)
+        if original is None:  # pragma: no cover
+            raise Exception("can not find original content on S3")
+
+        try:
+            for asset in _digest_content(original):
+                if asset[2] is None:  # skip it if it's not signed
+                    continue
+                upload_content_to_s3(name, v, asset, bucket, headers)
+        except:
+            raise
+        finally:
+            original.close()
+
+
+def upload_signed_content(content, name, version, freeze=False):
+    """Upload the signed content file to S3, return all the urls if succeeds
```

Comment:
This also uploads the original content to S3. We need a better function name and descriptive comments.